earth_enterprise/src/common/khThread.cpp

```diff
@@ -105,26 +122,32 @@ void* khThreadBase::thread_start(void *obj) {
     notify(NFY_WARN,
            "Internal error: Unknown exception escaped from thread");
   }
+  END_PERF_LOGGING(timingLogger);
```

Comment:
The time will be logged when the timingLogger object goes out of scope or when the END_PERF_LOGGING command is hit, so that covers exceptions as well.